drivers/SmartThings/sonos/src/init.lua

```diff
@@ -54,11 +52,7 @@ local function find_player_for_device(driver, device, should_continue)
   -- If the user doesn't provide a should_continue condition then we
   -- make one that just attempts it a handful of times.
   if type(should_continue) ~= "function" then
-    local attempts = 0
-    should_continue = function()
-      attempts = attempts + 1
-      return attempts <= DEFAULT_SSDP_RETRY_ATTEMPTS
-    end
+    should_continue = function() return false end
```

Comment:
It could be useful if other callers want to use interior retries instead of exterior retries. I'd like to leave it with the ability to take that parameter, even if it smells a bit of YAGNI, because I can already think of use-cases for it as we refactor this driver in the future to optimize it.